File: sept516.cpp
#include<bits/stdc++.h>
using namespace std
long a[1000000],ans[1000000],st[100000000];
vector<bool>is_prime(1000010,true);
void sieve()
{
	is_prime[0]=is_prime[1] = false;
	for(long i=2;i*i<1000010;i++)
	{
		if(is_prime[i]==true)
		{
			for(long j=i*i;j<1000010;j+=i)
			{
				if(is_prime==true)
					is_prime[j]=false;
			}
		}
	}
}
void seg_tree(long node,long s,long e)
{
	if(s==e)
		st[node]=lpd(a[s]);
	else
	{
		long mid = (s+e)/2;
		seg_tree(2*n + 1,s,mid);
		seg_tree(2*n + 2,mid+1,e);
		st[node] = max(st[2*node + 1],st[2*node + 2]);
	}
}
update(long node,long s,long e,long l,long r)
{
	if (lazy[node] != 0)
    {
        tree[node] += (e-s+1)*lazy[node];
        if (s != e)
        {
            lazy[node*2 + 1]   += lazy[node];
            lazy[node*2 + 2]   += lazy[node];
        }
        lazy[node] = 0;
    }
 
    // out of range
    if (s>e || s>r || e<l)
        return ;
    if (s>=l && e<=r)
    {
        // Add the difference to current node
        tree[node] += (e-s+1)*diff;
 
        // same logic for checking leaf node or not
        if (s != e)
        {
            lazy[node*2 + 1]   += diff;
            lazy[node*2 + 2]   += diff;
        }
        return;
    }
 
    // If not completely in rang, but overlaps, recur for
    // children,
    int mid = (s+e)/2;
    updateRangeUtil(node*2+1, s, mid, l, r);
    updateRangeUtil(node*2+2, mid+1, e, l, r);
 
    // And use the result of children calls to update this
    // node
    tree[node] = tree[node*2+1] + tree[node*2+2];
}
int main()
{
	long t,n,m,l,r;
	int type;
	cin>>t;
	sieve();
	while(t--)
	{
		long k=0;
		cin>>n>>m;
		for(long i=0;i<n;i++)
		{
			cin>>a[i];
		}
		seg_tree(0,0,n-1);
		for(long i=0;i<m;i++)
		{
			cin>>type>>l>>r;
			if(type==1)
			{
				ans[k++] = get(l,r);
			}
			else
			{
				update(0,0,n-1,l,r);
			}
		}
		for(long i=0;i<k;i++)
			cout<<ans[i]<<" ";
		cout<<endl;
	}
	return 0;
}

File: sept516.py
is_prime = [1]*(1000010)
lpd = [1]*(1000010)
is_prime[0]=is_prime[1]=0
def sieve():
	for i in range(1,1000001):
		if is_prime[i]==1:
			k = i*i
			while k<=1000000:
				if is_prime[k]==1:
					is_prime[k]=0
				k+=i

def lpdd():
	for i in range(2,1000001):
		if i%2==0:
			lpd[i]=2
		elif i%3==0:
			lpd[i]=3
		elif i%5==0:
			lpd[i]=5
		else:
			j=7
			while (j*j)<=i:
				if i%j==0:
					lpd[i]=j
					break
				elif i%(j+4)==0:
					lpd[i]=j+4
					break
				elif i%(j+6)==0:
					lpd[i]=j+6
					break
				elif i%(j+10)==0:
					lpd[i]=j+10
					break
				elif i%(j+12)==0:
					lpd[i]=j+12
					break
				elif i%(j+16)==0:
					lpd[i]=j+16
					break
				elif i%(j+22)==0:
					lpd[i]=j+22
					break
				elif i%(j+24)==0:
					lpd[i]=j+24
					break
				j+=30
			if j*j > i:
				lpd[i] = i

sieve()
lpdd()
while j<=t:
	z = 0
	n = [int(x) for x in input().split()]
	#m = int(input())
	result = [0]*(n[1])
	a = [int(x) for x in input().split()]
	for i in range(0,n[1]):
		b = [int(y) for y in input().split()]
		if b[0]==0:
			update(b[1],b[2],a)
		else:
			result[z] = get(b[1],b[2],a)
			#print(result[z])
			z+=1
	result = result[0:z]
	print(*result)
	j+=1

